notes/thread-design.md

```diff
@@ -0,0 +1,142 @@
+# FDB Client Threads
+
+The default configuration of erlfdb includes the creation and management of a single OS thread that is responsible for all database work created by processes in the Erlang VM. As your workload increases in size, you will need to configure the erlfdb client for horizontal scaling. Otherwise, a sufficiently sized system with a sufficiently sized workload can saturate the single client thread with work, which will lead to high latency on individual requests, and a ceiling on throughput.
+
+This page will guide you through the configuration necessary to scale the FDB client to several worker threads on a single BEAM VM.
+
+## Network options configuration
+
+The configuration will be done via the erlfdb `network_options` environment variable. In erlfdb >= 0.3, defaults are chosen that are meant to ease the configuration burden. The options are described in detail below, but the main takeaway is the following:
+
+> #### Tip {: .tip}
+>
+> In order to double your client's maximum throughput, set
+> `client_threads_per_version` to 2.
+
+```erlang
+[
+    {erlfdb, [
+        {network_options, [
+            {client_threads_per_version, 2}
+        ]}
+    ]}
+].
+```
+
+Please read on for details about the scaling implementation.
+
+### Version 0.3 and above
+
+In versions 0.3 and above, the env var `network_options` defaults to
+
+```erlang
+[
+    {erlfdb, [
+        {network_options, [
+            % `ignore_external_client_failures`:
+            % Optional: Allows your client to prepare fdbserver version
+            % upgrades. When a future client library fails to load, it's not a
+            % failure condition, because we know it will be valid at some
+            % point in the future.
+            {ignore_external_client_failures, true},
```

Comment:
I see you updated some tests - do we have any tests that measure whether erlfdb can cope with clients connected to two running FDB versions at the same time?

Reply:
No, we don't have such tests yet